DOCS: Add ruff section to the dev guide docs (#5701)
SciTools/iris: docs/src/developers_guide/contributing_ci_tests.rst

  Append to the ``ignore-words-list`` option any **valid words** that are
  considered **not** a typo and should **not** be corrected by `codespell`_.

ruff
----
As of **Iris 3.8** `ruff`_ has been adopted to ensure our codebase is using best

> practice.  rruff is configured in the `Iris`_ GitHub repository using  

Reviewer: ```suggestion
practice.  `ruff`_ is configured in the `Iris`_ GitHub repository using  
```

`.pre-commit-config.yaml`_.  

> You can install and run `ruff`_ into your development **iris-dev** conda environment

Reviewer: ```suggestion
You can install and run `ruff`_ in your development **iris-dev** conda environment
```

via::

   conda activate iris-dev
   pip install ruff
   cd iris
   ruff .

.. note::

  The `ruff`_ ``pre-commit`` hook checks for compliance of the whole codebase.
  This hook is configured in the ``[tool.ruff]`` section
  of the ``pyproject.toml`` file.

>   Edit the ``.ruff.toml`` file to add any rules to ignore if appropriate.  We

Reviewer: ```suggestion
  Edit the ``.ruff.toml`` file to include any *temporary* rules to be ignored. Edit the ``pyproject.toml`` to include any *permanent* rules to be ignored. We
```

>   aim to be ruff compliant where at all possible.

Reviewer: ```suggestion
  aim to be fully `ruff`_ compliant as possible.
```
